tests/benchmarks/object.cpp

```diff
@@ -209,30 +261,415 @@ TEST_CASE("Benchmark object", "[benchmark]") {
 
     SECTION("change notifications reporting") {
         auto table = r->read_group().get_table("class_person");
-        Results result(r, *table);
+        Results result(r, table);
         size_t num_calls = 0;
         size_t num_insertions = 0;
+        size_t num_deletions = 0;
+        size_t num_modifications = 0;
         auto token = result.add_notification_callback([&](CollectionChangeSet c, std::exception_ptr) {
             num_insertions += c.insertions.count();
+            num_deletions += c.deletions.count();
+            num_modifications += c.modifications_new.count();
             ++num_calls;
         });
 
         advance_and_notify(*r);
-        int64_t pk = 0;
+        ObjectSchema person_schema = *r->schema().find("person");
+        constexpr size_t num_objects = 1000;
+
         BENCHMARK_ADVANCED("create notifications")(Catch::Benchmark::Chronometer meter) {
-            std::stringstream name;
-            name << "person_" << pk++;
-            AnyDict person {
-                {"name", name.str()},
-                {"age", pk},
+            r->begin_transaction();
+            result.clear();
+            r->commit_transaction();
+            advance_and_notify(*r);
+            num_insertions = 0;
+            num_modifications = 0;
+            num_deletions = 0;
+
+            r->begin_transaction();
+            for (size_t i = 0; i < num_objects; ++i) {
+                std::stringstream name;
+                name << "person_" << i;
+                AnyDict person {
+                    {"name", name.str()},
+                    {"age", static_cast<int64_t>(i)},
+                };
+                Object::create(d, r, person_schema, Any(person), CreatePolicy::ForceCreate);
+            }
+            r->commit_transaction();
+            meter.measure([&r] {
+                on_change_but_no_notify(*r);
+            });
+            r->notify();
+            REQUIRE(num_insertions == num_objects);
+            REQUIRE(num_modifications == 0);
+            REQUIRE(num_deletions == 0);
+            REQUIRE(result.size() == num_objects);
+        };
+
+        r->begin_transaction();
+        result.clear();
+        r->commit_transaction();
+        advance_and_notify(*r);
+        num_calls = 0;
+
+        BENCHMARK_ADVANCED("delete notifications")(Catch::Benchmark::Chronometer meter) {
```

Comment:
Some other things that'd be interesting to benchmark are transactions involving a lot of tables and transactions where deep change checking come into play (i.e. where a linked-to table is modified).